api/src/main/java/io/smallrye/stork/ServiceMetadata.java

```diff
@@ -0,0 +1,23 @@
+package io.smallrye.stork;
+
+import java.util.Map;
+
+public class ServiceMetadata {
+
+    private final Map<String, String> labels;
+    //TODO rename
+    private final Map<String, Object> metadata;
+
+    public ServiceMetadata(Map<String, String> labels, Map<String, Object> metadata) {
```

Comment:
What's the rationale behind these two lists? Aren't labels a list of strings that can be stored in metadata, or keys with a specific "singleton" value?

Reply:
In K8s labels are key-value pairs we were leaning towards making a special case for them, the `Map<String, String> labels`. The other stuff is not common to all serviceDiscovery implementations so it seemed better to me to wrap them in the Map<String,Object> at the moment and see how it goes